platform-sdk/swirlds-config-processor/src/main/java/com/swirlds/config/processor/ConfigRecordParser.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.config.processor;
+
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.tools.FileObject;
+import org.jboss.forge.roaster.Roaster;
+import org.jboss.forge.roaster.model.Annotation;
+import org.jboss.forge.roaster.model.JavaRecord;
+import org.jboss.forge.roaster.model.JavaRecordComponent;
+import org.jboss.forge.roaster.model.JavaType;
+
+public class ConfigRecordParser {
+
+    private ConfigRecordParser() {}
+
+    @NonNull
+    private static Map<String, String> getJavaDocParams(@NonNull final JavaType<?> type) {
+        Objects.requireNonNull(type, "type must not be null");
+        final Map<String, String> paramDoc = new HashMap<>();
+        type.getJavaDoc().getTags(ConfigProcessorConstants.JAVADOC_PARAM).forEach(tag -> {
+            int split = tag.getValue().indexOf(" ");
+            paramDoc.put(
+                    tag.getValue().substring(0, split).trim(),
+                    tag.getValue().substring(split + 1).trim());
+        });
+        return Collections.unmodifiableMap(paramDoc);
+    }
+
+    @NonNull
+    public static ConfigDataRecordDefinition parse(@NonNull final FileObject javaSourceFile) throws IOException {
```

Comment:
Please add javadoc to public method (nice to have for others as well, but we definitely want them for public methods)